### PR TITLE
Cleanup versions

### PR DESCRIPTION
**Pull request by ZappoMan**
This just cleans up the entities versioning info from PacketHeaders.h there are no functional changes to this.

Note to reviewer: you might want to review this with ?w=1 to only see the non-whitespace changes... there are several chunks of code that have been removed from inside of "if statements" that were always true... so it's easier to see the actual change by ignoring those lines which only change indent.

Test plan:
- make sure content loads properly from existing domains
- in particular, create/test particle systems

### CODE REVIEW

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -355,14 +355,6 @@ int EntityItem::expectedBytes() {
 
 // clients use this method to unpack FULL updates from entity-server
 int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLeftToRead, ReadBitstreamToTreeParams& args) {
-    if (args.bitstreamVersion < VERSION_ENTITIES_SUPPORT_SPLIT_MTU) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: no longer supported bit stream... everything is always greater than this, so this is always false

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -588,34 +580,32 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
 
     // Newer bitstreams will have a last simulated and a last updated value
     quint64 lastSimulatedFromBufferAdjusted = now;
-    if (args.bitstreamVersion >= VERSION_ENTITIES_HAS_LAST_SIMULATED_TIME) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/EntityItem.cpp [hunk 3]
```diff
@@ -817,20 +807,16 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
     READ_ENTITY_PROPERTY(PROP_LOCKED, bool, updateLocked);
     READ_ENTITY_PROPERTY(PROP_USER_DATA, QString, setUserData);
 
-    if (args.bitstreamVersion >= VERSION_ENTITIES_HAS_MARKETPLACE_ID) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/EntityItem.cpp [hunk 4]
```diff
@@ -817,20 +807,16 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
...
-        READ_ENTITY_PROPERTY(PROP_MARKETPLACE_ID, QString, setMarketplaceID);
-    }
-    if (args.bitstreamVersion >= VERSION_ENTITIES_HAS_CERTIFICATE_PROPERTIES) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/EntityItem.cpp [hunk 5]
```diff
@@ -858,10 +844,6 @@ int EntityItem::readEntityDataFromBuffer(const unsigned char* data, int bytesLef
     // NOTE: we had a bad version of the stream that we added stream data after the subclass. We can attempt to recover
     // by doing this parsing here... but it's not likely going to fully recover the content.
     //
-    // TODO: Remove this code once we've sufficiently migrated content past this damaged version
-    if (args.bitstreamVersion == VERSION_ENTITIES_HAS_MARKETPLACE_ID_DAMAGED) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItem.cpp]: everything is always greater than this, so this is always FALSE - so I removed the code

File: libraries/entities/src/EntityTree.h [hunk 6]
```diff
@@ -89,8 +89,6 @@ class EntityTree : public Octree, public SpatialParentTree {
     // own definition. Implement these to allow your octree based server to support editing
     virtual bool getWantSVOfileVersions() const override { return true; }
     virtual PacketType expectedDataPacketType() const override { return PacketType::EntityData; }
-    virtual bool canProcessVersion(PacketVersion thisVersion) const override
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTree.h]: dead code

File: libraries/entities/src/EntityTree.h [hunk 7]
```diff
@@ -111,9 +109,6 @@ class EntityTree : public Octree, public SpatialParentTree {
     virtual void releaseSceneEncodeData(OctreeElementExtraEncodeData* extraEncodeData) const override;
     virtual bool mustIncludeAllChildData() const override { return false; }
 
-    virtual bool versionHasSVOfileBreaks(PacketVersion thisVersion) const override
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTree.h]: dead code

File: libraries/entities/src/EntityTypes.cpp [hunk 8]
```diff
@@ -102,12 +102,6 @@ EntityItemPointer EntityTypes::constructEntityItem(EntityType entityType, const
 EntityItemPointer EntityTypes::constructEntityItem(const unsigned char* data, int bytesToRead,
             ReadBitstreamToTreeParams& args) {
 
-    if (args.bitstreamVersion < VERSION_ENTITIES_SUPPORT_SPLIT_MTU) {
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTypes.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/LightEntityItem.cpp [hunk 9]
```diff
@@ -175,35 +175,12 @@ int LightEntityItem::readEntitySubclassDataFromBuffer(const unsigned char* data,
     int bytesRead = 0;
     const unsigned char* dataAt = data;
 
-    if (args.bitstreamVersion < VERSION_ENTITIES_LIGHT_HAS_INTENSITY_AND_COLOR_PROPERTIES) {
```
Comment (ZappoMan) [inline libraries/entities/src/LightEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ModelEntityItem.cpp [hunk 10]
```diff
@@ -111,38 +111,19 @@ int ModelEntityItem::readEntitySubclassDataFromBuffer(const unsigned char* data,
 
     READ_ENTITY_PROPERTY(PROP_COLOR, rgbColor, setColor);
     READ_ENTITY_PROPERTY(PROP_MODEL_URL, QString, setModelURL);
-    if (args.bitstreamVersion < VERSION_ENTITIES_HAS_COLLISION_MODEL) {
```
Comment (ZappoMan) [inline libraries/entities/src/ModelEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ModelEntityItem.cpp [hunk 11]
```diff
@@ -111,38 +111,19 @@ int ModelEntityItem::readEntitySubclassDataFromBuffer(const unsigned char* data,
...
-        setCompoundShapeURL("");
-    } else {
-        READ_ENTITY_PROPERTY(PROP_COMPOUND_SHAPE_URL, QString, setCompoundShapeURL);
-    }
-
-    // Because we're using AnimationLoop which will reset the frame index if you change it's running state
-    // we want to read these values in the order they appear in the buffer, but call our setters in an
-    // order that allows AnimationLoop to preserve the correct frame rate.
-    if (args.bitstreamVersion < VERSION_ENTITIES_ANIMATION_PROPERTIES_GROUP) {
```
Comment (ZappoMan) [inline libraries/entities/src/ModelEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ModelEntityItem.cpp [hunk 12]
```diff
@@ -111,38 +111,19 @@ int ModelEntityItem::readEntitySubclassDataFromBuffer(const unsigned char* data,
...
-        READ_ENTITY_PROPERTY(PROP_ANIMATION_URL, QString, setAnimationURL);
-        READ_ENTITY_PROPERTY(PROP_ANIMATION_FPS, float, setAnimationFPS);
-        READ_ENTITY_PROPERTY(PROP_ANIMATION_FRAME_INDEX, float, setAnimationCurrentFrame);
-        READ_ENTITY_PROPERTY(PROP_ANIMATION_PLAYING, bool, setAnimationIsPlaying);
-    }
-
+    READ_ENTITY_PROPERTY(PROP_COMPOUND_SHAPE_URL, QString, setCompoundShapeURL);
     READ_ENTITY_PROPERTY(PROP_TEXTURES, QString, setTextures);
 
-    if (args.bitstreamVersion < VERSION_ENTITIES_ANIMATION_PROPERTIES_GROUP) {
```
Comment (ZappoMan) [inline libraries/entities/src/ModelEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 13]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
     const unsigned char* dataAt = data;
 
     READ_ENTITY_PROPERTY(PROP_COLOR, rgbColor, setColor);
-    // Because we're using AnimationLoop which will reset the frame index if you change it's running state
-    // we want to read these values in the order they appear in the buffer, but call our setters in an
-    // order that allows AnimationLoop to preserve the correct frame rate.
-    if (args.bitstreamVersion < VERSION_ENTITIES_ANIMATION_PROPERTIES_GROUP) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 14]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-    } else {
-        READ_ENTITY_PROPERTY(PROP_EMITTING_PARTICLES, bool, setIsEmitting);
-    }
-
+    READ_ENTITY_PROPERTY(PROP_EMITTING_PARTICLES, bool, setIsEmitting);
     READ_ENTITY_PROPERTY(PROP_SHAPE_TYPE, ShapeType, setShapeType);
     READ_ENTITY_PROPERTY(PROP_MAX_PARTICLES, quint32, setMaxParticles);
     READ_ENTITY_PROPERTY(PROP_LIFESPAN, float, setLifespan);
     READ_ENTITY_PROPERTY(PROP_EMIT_RATE, float, setEmitRate);
-    if (args.bitstreamVersion < VERSION_ENTITIES_PARTICLE_ELLIPSOID_EMITTER) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 15]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-        // OLD PROP_EMIT_VELOCITY FAKEOUT
-        SKIP_ENTITY_PROPERTY(PROP_EMIT_SPEED, glm::vec3);
-    }
 
-    if (args.bitstreamVersion >= VERSION_ENTITIES_PARTICLE_MODIFICATIONS) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 16]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-        READ_ENTITY_PROPERTY(PROP_EMIT_ACCELERATION, glm::vec3, setEmitAcceleration);
-        READ_ENTITY_PROPERTY(PROP_ACCELERATION_SPREAD, glm::vec3, setAccelerationSpread);
-        READ_ENTITY_PROPERTY(PROP_PARTICLE_RADIUS, float, setParticleRadius);
-        READ_ENTITY_PROPERTY(PROP_TEXTURES, QString, setTextures);
-        if (args.bitstreamVersion < VERSION_ENTITIES_PARTICLE_ELLIPSOID_EMITTER) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always false

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 17]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-    } else {
-        // OLD PROP_EMIT_ACCELERATION FAKEOUT
-        SKIP_ENTITY_PROPERTY(PROP_PARTICLE_RADIUS, float);
-        // OLD PROP_ACCELERATION_SPREAD FAKEOUT
-        SKIP_ENTITY_PROPERTY(PROP_PARTICLE_RADIUS, float);
-        READ_ENTITY_PROPERTY(PROP_PARTICLE_RADIUS, float, setParticleRadius);
-        READ_ENTITY_PROPERTY(PROP_TEXTURES, QString, setTextures);
-    }
-
-    if (args.bitstreamVersion >= VERSION_ENTITIES_PARTICLE_RADIUS_PROPERTIES) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 18]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-        READ_ENTITY_PROPERTY(PROP_RADIUS_SPREAD, float, setRadiusSpread);
-        READ_ENTITY_PROPERTY(PROP_RADIUS_START, float, setRadiusStart);
-        READ_ENTITY_PROPERTY(PROP_RADIUS_FINISH, float, setRadiusFinish);
-    }
-
-    if (args.bitstreamVersion >= VERSION_ENTITIES_PARTICLE_COLOR_PROPERTIES) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 19]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-        READ_ENTITY_PROPERTY(PROP_COLOR_SPREAD, xColor, setColorSpread);
-        READ_ENTITY_PROPERTY(PROP_COLOR_START, xColor, setColorStart);
-        READ_ENTITY_PROPERTY(PROP_COLOR_FINISH, xColor, setColorFinish);
-        READ_ENTITY_PROPERTY(PROP_ALPHA, float, setAlpha);
-        READ_ENTITY_PROPERTY(PROP_ALPHA_SPREAD, float, setAlphaSpread);
-        READ_ENTITY_PROPERTY(PROP_ALPHA_START, float, setAlphaStart);
-        READ_ENTITY_PROPERTY(PROP_ALPHA_FINISH, float, setAlphaFinish);
-    }
-
-    if (args.bitstreamVersion >= VERSION_ENTITIES_PARTICLE_ELLIPSOID_EMITTER) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/entities/src/ParticleEffectEntityItem.cpp [hunk 20]
```diff
@@ -464,76 +464,40 @@ int ParticleEffectEntityItem::readEntitySubclassDataFromBuffer(const unsigned ch
...
-        READ_ENTITY_PROPERTY(PROP_EMIT_ORIENTATION, glm::quat, setEmitOrientation);
-        READ_ENTITY_PROPERTY(PROP_EMIT_DIMENSIONS, glm::vec3, setEmitDimensions);
-        READ_ENTITY_PROPERTY(PROP_EMIT_RADIUS_START, float, setEmitRadiusStart);
-        READ_ENTITY_PROPERTY(PROP_POLAR_START, float, setPolarStart);
-        READ_ENTITY_PROPERTY(PROP_POLAR_FINISH, float, setPolarFinish);
-        READ_ENTITY_PROPERTY(PROP_AZIMUTH_START, float, setAzimuthStart);
-        READ_ENTITY_PROPERTY(PROP_AZIMUTH_FINISH, float, setAzimuthFinish);
-    }
-
-    if (args.bitstreamVersion >= VERSION_ENTITIES_PARTICLES_ADDITIVE_BLENDING) {
```
Comment (ZappoMan) [inline libraries/entities/src/ParticleEffectEntityItem.cpp]: everything is always greater than this, so this is always true

File: libraries/octree/src/Octree.cpp [hunk 21]
```diff
@@ -1721,146 +1721,15 @@ bool Octree::readFromStream(uint64_t streamLength, QDataStream& inputStream, con
...
+        qCWarning(octree) << "Reading from binary SVO no longer supported";
+        return false;
     } else {
         qCDebug(octree) << "Reading from JSON SVO Stream length:" << streamLength;
         return readJSONFromStream(streamLength, inputStream, marketplaceID);
     }
 }
 
 
-bool Octree::readSVOFromStream(uint64_t streamLength, QDataStream& inputStream) {
```
Comment (ZappoMan) [inline libraries/octree/src/Octree.cpp]: We've obsoleted this feature a while ago... removing the dead code

File: libraries/entities/src/EntityItem.cpp (hunk 1 above)
Comment (jherico) [reply libraries/entities/src/EntityItem.cpp]: shouldn't we still have _some_ guard so that bad packets are ignored rather than potentially crash?

**Comment by YuppyPlays** [comment]
Awaiting build for QA.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4398/HighFidelity-Beta-PR11668-aad91edba202a7bdf6e8cc016b08e08ed387eaa4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4398/HighFidelity-Beta-PR11668-aad91edba202a7bdf6e8cc016b08e08ed387eaa4.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

make sure content loads properly from existing domains **Domain mismatch with places like Dev-welcome, Welcome, Wikiplanet, etc..**
in particular, create/test particle systems **Can't get into other domains to test**

**Fail** for now as I cannot continue for testing.

**Comment by ZappoMan** [comment]
@YuppyPlays - yeah this is a protocol change. so you need to test with content in the sandbox for this build...

**Comment by YuppyPlays** [comment]
@ZappoMan Apologies, I misunderstood with the wording of existing domains, do I need to have a domain  in master and bring over the sandbox data into PR to ensure it all loads correctly? Or just verify sandbox loads in general within PR, then create and test the particle systems?

**Comment by ZappoMan** [comment]
@YuppyPlays - if content loads in general - then it should be good... particles are the one special case, because they had a bunch of special case code that I removed (code that shouldn't have been executing, but it's now gone)...

**Comment by YuppyPlays** [comment]
@ZappoMan I believe the sandbox did load correctly, I'll quicklyre-check and check regarding particles.

Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

make sure content loads properly from existing domains **Check, Domain sandbox loads correctly**
in particular, create/test particle systems **Tested particle effects in CREATE and fiddled around with settings, works and renders as usual**

**Pull request merged**